src/anndata/_core/sparse_dataset.py

```diff
@@ -607,6 +607,11 @@ def sparse_dataset(group: GroupStorageType) -> CSRDataset | CSCDataset:
     >>> import scanpy as sc
     >>> import h5py
     >>> from anndata.experimental import sparse_dataset, read_elem
+    >>> import scipy
+    >>> import pytest
+    >>> from packaging.version import Version
+    >>> if Version(scipy.__version__) >= Version("1.14.0rc1"):
+    ...     pytest.xfail("scipy.sparse string formatting changed in 1.14.0rc1 and up")
```

Comment:
Isaac said that he doesn‘t like using pytest stuff visibly in docstrings.

You could e.g. use `# doctest: +ELLIPSIS` and keep the parts that stay the same.

Reply:
I was trying to fiddle around with this but I couldn't figure out how to get pytest to use a custom doctest runner: https://stackoverflow.com/a/55480919/8060591

Reply:
Why would that be necessary? you just do this instead of what you’re doing

```diff
->>> X[100:200]  # doctest: +NORMALIZE_WHITESPACE
-<100x765 sparse matrix of type '<class 'numpy.float32'>'
-    with 25003 stored elements in Compressed Sparse Row format>
+>>> X[100:200]  # doctest: +ELLIPSIS
+<...sparse matrix of...float32...with 25003 stored elements...>
```

Reply:
Well it would allow us to match exactly if I could get it work.  But this is also fine.